assets/scss/local.scss

```diff
@@ -21,31 +21,35 @@
   width: 20em;
 }
 
-// Custom grid and border for goals section of overview page
+// Custom column with vertical border
 
-.govuk-grid {
-  gap: 20px;
-  display: flex;
+.app-table-column-with-right-border {
+  border-right: 1px solid $govuk-border-colour;
+  display: inline-block;
 }
 
-.govuk-grid-item {
-  padding: 16px;
-  text-align: center;
-}
+.overview-page-goal-summary-counts {
+  display: grid;
+  grid-auto-columns: minmax(0, 1fr);
```

Comment:
auto-columns rather than specifying 3 columns (as we had in the original PR) fixes the problem of whether there are 2 or 3 (or more 🤷‍♂️ ) columns, and they all get nicely spaced out 👍